src/agent/rustjail/src/cgroups/fs/mod.rs

```diff
@@ -27,1120 +43,848 @@ macro_rules! sl {
-pub const MEMORY_LIMIT: &'static str = "memory.limit_in_bytes";
-pub const MEMORY_SOFT_LIMIT: &'static str = "memory.soft_limit_in_bytes";
-pub const MEMSW_LIMIT: &'static str = "memory.memsw.limit_in_bytes";
-pub const KMEM_LIMIT: &'static str = "memory.kmem.limit_in_bytes";
-pub const KMEM_TCP_LIMIT: &'static str = "memory.kmem.tcp.limit_in_bytes";
-pub const SWAPPINESS: &'static str = "memory.swappiness";
-pub const OOM_CONTROL: &'static str = "memory.oom_control";
-pub const PIDS_MAX: &'static str = "pids.max";
-pub const BLKIO_WEIGHT: &'static str = "blkio.weight";
-pub const BLKIO_LEAF_WEIGHT: &'static str = "blkio.leaf_weight";
-pub const BLKIO_WEIGHT_DEVICE: &'static str = "blkio.weight_device";
-pub const BLKIO_LEAF_WEIGHT_DEVICE: &'static str = "blkio.leaf_weight_device";
-pub const BLKIO_READ_BPS_DEVICE: &'static str = "blkio.throttle.read_bps_device";
-pub const BLKIO_WRITE_BPS_DEVICE: &'static str = "blkio.throttle.write_bps_device";
-pub const BLKIO_READ_IOPS_DEVICE: &'static str = "blkio.throttle.read_iops_device";
-pub const BLKIO_WRITE_IOPS_DEVICE: &'static str = "blkio.throttle.write_iops_device";
-pub const NET_CLS_CLASSID: &'static str = "net_cls.classid";
-pub const NET_PRIO_IFPRIOMAP: &'static str = "net_prio.ifpriomap";
-
-pub const CPU_STAT: &'static str = "cpu.stat";
-pub const CPUACCT_STAT: &'static str = "cpuacct.stat";
-pub const NANO_PER_SECOND: u64 = 1000000000;
-pub const CPUACCT_USAGE: &'static str = "cpuacct.usage";
-pub const CPUACCT_PERCPU: &'static str = "cpuacct.usage_percpu";
-pub const MEMORY_STAT: &'static str = "memory.stat";
-pub const MEM_USAGE: &'static str = "usage_in_bytes";
-pub const MEM_MAX_USAGE: &'static str = "max_usage_in_bytes";
-pub const MEM_FAILCNT: &'static str = "failcnt";
-pub const MEM_LIMIT: &'static str = "limit_in_bytes";
-pub const MEM_HIERARCHY: &'static str = "memory.use_hierarchy";
-pub const PIDS_CURRENT: &'static str = "pids.current";
-pub const BLKIO_SECTORS: &'static str = "blkio.sectors_recursive";
-pub const BLKIO_IO_SERVICE_BYTES: &'static str = "blkio.io_service_bytes_recursive";
-pub const BLKIO_IO_SERVICED: &'static str = "blkio.io_serviced_recursive";
-pub const BLKIO_IO_QUEUED: &'static str = "blkio.io_queued_recursive";
-pub const BLKIO_IO_SERVICE_TIME: &'static str = "blkio.io_service_time_recursive";
-pub const BLKIO_IO_WAIT_TIME: &'static str = "blkio.io_wait_time_recursive";
-pub const BLKIO_IO_MERGED: &'static str = "blkio.io_merged_recursive";
-pub const BLKIO_TIME: &'static str = "blkio.time_recursive";
-pub const BLKIO_THROTTLE_IO_SERVICE_BYTES: &'static str = "blkio.throttle.io_service_bytes";
-pub const BLKIO_THROTTLE_IO_SERVICED: &'static str = "blkio.throttle.io_serviced";
+        let res = &mut cgroups::Resources::default();
 
-lazy_static! {
-    pub static ref CLOCK_TICKS: f64 = {
-        let n = unsafe { libc::sysconf(libc::_SC_CLK_TCK) };
+        // cpuset
+        if r.cpu.is_some() {
+            info!(sl!(), "cgroup manager set cpu !!!");
```

Comment:
Sorry, it have something about debug purpose. I'll delete it.

Reply:
I think it is worthwhile to leave the `info` calls, as long as they are not overly noisy. Just remove the `!!!` to avoid people thinking there is anything nasty going on.